src/utils/mod.rs

```diff
@@ -36,3 +36,74 @@ pub fn word_to_hex(w: &Word) -> Result<String, fmt::Error> {
 
     Ok(s)
 }
+
+/// Renders an array of bytes as hex into a String.
+pub fn bytes_to_hex_string<const N: usize>(data: [u8; N]) -> String {
+    let mut s = String::with_capacity(N + 2);
+
+    s.push_str("0x");
+    for byte in data.iter() {
+        write!(s, "{byte:02x}").expect("formatting hex failed");
+    }
+
+    s
+}
+
+#[derive(Debug)]
+pub enum HexParseError {
+    InvalidLength { expected: usize, got: usize },
+    MissingPrefix,
+    InvalidChar,
+}
+
+impl Display for HexParseError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            HexParseError::InvalidLength { expected, got } => {
+                write!(f, "Hex encoded RpoDigest must have length 66, including the 0x prefix. expected {expected} got {got}")
+            }
+            HexParseError::MissingPrefix => {
+                write!(f, "Hex encoded RpoDigest must start with 0x prefix")
+            }
+            HexParseError::InvalidChar => {
+                write!(f, "Hex encoded RpoDigest must contain characters [a-zA-Z0-9]")
+            }
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for HexParseError {}
+
+/// Parses a hex string into an array of bytes of known size.
+pub fn hex_to_bytes<const N: usize>(value: &str) -> Result<[u8; N], HexParseError> {
```

Comment:
We have a similar utility downstream in `miden-vm`. Would be good if we can purge that and leverage this. https://github.com/0xPolygonMiden/miden-vm/blob/105ee9e241fafeac97fbcd14861b2eaed5548e1e/assembly/src/ast/parsers/labels.rs#L77-L93